go-ethereum-guides/write-basic-quiz-dapp-in-go/write-basic-quiz-dapp-in-go.md

```diff
@@ -0,0 +1,1069 @@
+            return _updateLeaderBoard(_answer == _ans);
+        }
+
+        function _updateLeaderBoard(bool ok) internal returns (bool){
+            _leaderBoard[msg.sender] = ok;
+            return true;
+        }
+
+        function checkBoard() public view returns (bool){
+            return _leaderBoard[msg.sender];
+        }
+    }
+    ```
+
+We'll cover what our smart contract code does briefly; 
+for more information about writing smart contracts in Solidity, see 
+[this guide](https://beta.kauri.io/article/124b7db1d0cf4f47b414f8b13c9d66e2/v5).
+
+In our contract above, we're:
+
+1. Setting the data types that we want to store on the contract.
+    - `string public question`: 
+    Stores the question that we want to ask the user. 
+    Setting this as `public` has Solidity automatically
+    generate a getter function for it when the contract compiles.
+    This allows us to read this variable's contents with a 
+    `contractInstance.question()` method.
+    Because getters don't invoke code execution on the EVM,
+    they don't cost gas to run.
+    - `bytes32 internal _answer`: Stores the answer to our question.
+    We've set an `internal` modifier, which means that
+    this variable can only be accessed from within this contract.
+    - `mapping (address=>bool) internal _leaderBoard`: 
+    Stores a hash map of user accounts and a boolean value 
+    that tells us whether a given account 
+    has answered the question correctly.
+    We've also set this state variable as `internal`
+    to prevent external callers from modifying its contents.
+2. The `constructor` is called when we deploy the contract.
+Here, we give it a question `_qn` and an answer `_ans`.
+    - We take `_qn` as a string because we mean for 
+    it to be easily readable by anyone interacting with the contract.
+    - Our answer `_ans` is set as a fixed slice of 32 bytes (`bytes32`)
+    because we want to store it as a `keccak256` hash. 
+    Hashing the value of `_ans` obscures it,
+    making it unreadable in the contract source or
+    the transaction logs.
+3. `sendAnswer()` allows us to send an answer to the contract.
+The answer sent to the contract must be a 32 byte
+keccak256 hash, which we compare to the value of 
+`_answer` stored on the contract.
+If the values match, we update our `leaderBoard` to show that
+the account that makes this function call has answered correctly.
+4. `_updateLeaderBoard()` takes a true/false value and sets
+the entry on the `_leaderBoard` mapping for our user's account to that value.
+It's an `internal` function, which prevents external callers
+from arbitarily modifying the `_leaderBoard` mapping.
+5. `checkBoard()` lets the user check the entry on the `_leaderBoard` mapping
+for their account, and whether they have been recorded as answering the question
+correctly. This function doesn't require EVM code execution as it's just
+returning the contents a state variable. It doesn't cost gas, and can be set 
```

Comment:
@zeddee I think there's a small typo here and I'm not quite sure what you're trying to say

Reply:
How about:

```
`checkBoard()` checks a user check if they've been recorded by the contract as having answered the question correctly.
```

Reply:
yep 👍